tartiflette/schema/schema.py

```diff
@@ -582,8 +631,217 @@ def _validate_directive_implementation(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLEnumTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLEnumType, "ENUM"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                values = [x.name for x in extended.values]
+                for value in extension.values:
+                    if value.name in values:
+                        errors.append(
+                            f"Can't add < {value.name} > Value "
+                            f"to < {extension.name} > "
+                            f"ENUM, cause value already exists."
+                        )
+
+                errors.extend(
+                    _validate_extension_directives(extension, extended, "ENUM")
+                )
+
+        return errors
+
+    def _validate_object_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLObjectTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLObjectType, "TYPE"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                for field in extension.fields:
+                    if field in extended.implemented_fields:
+                        errors.append(
+                            f"Can't add Field < {field} > to "
+                            f"TYPE < {extended.name} > "
+                            f"cause field already exists."
+                        )
+
+                for interface in extension.interfaces:
+                    if interface in extended.interfaces_names:
+                        errors.append(
+                            f"Can't add Interface < {interface} > "
+                            f"to TYPE < {extended.name} > "
+                            f"cause Interface already exists."
+                        )
+
+                errors.extend(
+                    _validate_extension_directives(extension, extended, "TYPE")
+                )
+
+        return errors
+
+    def _validate_union_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLUnionTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLUnionType, "UNION"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                for typ in extension.types:
```

Comment:
Using `already_defined` function, this could be rewrite as:
```python
errors.extend(
    already_defined(
        extended.types,
        extension.types,
        extended.name,
        "Can't add possible type < {existing_value} > to union < {extended_name} > cause possible type already exists.",
    )
)
```